File: ems_backend/internal/domain/audit_log/entities/audit_log.go
package entities

import "time"

// AuditLog - 審計日誌
type AuditLog struct {
	ID           uint                   `json:"id"`
	MemberID     uint                   `json:"member_id"`
	RoleID       uint                   `json:"role_id"`
	Action       string                 `json:"action"`        // CREATE, UPDATE, DELETE, LOGIN, LOGOUT等
	ResourceType string                 `json:"resource_type"` // MENU, ROLE, POWER, USER等
	ResourceID   *uint                  `json:"resource_id"`   // 受影響資源的ID（可為空）
	Details      map[string]interface{} `json:"details"`       // 額外的上下文信息
	IPAddress    string                 `json:"ip_address"`
	UserAgent    string                 `json:"user_agent"`
	Status       string                 `json:"status"`        // SUCCESS 或 FAILURE
	ErrorMessage string                 `json:"error_message"` // 如果失敗，記錄錯誤信息
	CreateTime   time.Time              `json:"create_time"`
}

// AuditLogFilter - 審計日誌查詢過濾器
type AuditLogFilter struct {
	MemberID     *uint     `json:"member_id"`
	RoleID       *uint     `json:"role_id"`
	Action       string    `json:"action"`
	ResourceType string    `json:"resource_type"`
	Status       string    `json:"status"`
	StartTime    time.Time `json:"start_time"`
	EndTime      time.Time `json:"end_time"`
	Limit        int       `json:"limit"`
	Offset       int       `json:"offset"`
}

File: ems_frontend/src/api/auditLog.ts
import axios from "../helper/axios";

// Types
export interface AuditLog {
  id: number;
  member_id: number;
  role_id: number;
  action: string;
  resource_type: string;
  resource_id?: number;
  details: Record<string, any>;
  ip_address: string;
  user_agent: string;
  status: string;
  error_message: string;
  create_time: string;
}

export interface AuditLogQueryParams {
  member_id?: number;
  role_id?: number;
  action?: string;
  resource_type?: string;
  status?: string;
  start_time?: string;
  end_time?: string;
  limit?: number;
  offset?: number;
}

export interface AuditLogListResponse {
  total: number;
  logs: AuditLog[];
}

// API functions
export const fetchAuditLogs = async (params?: AuditLogQueryParams) => {
  try {
    const response = await axios.get("/audit-logs", { params });
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const fetchAuditLogById = async (id: number) => {
  try {
    const response = await axios.get(`/audit-logs/${id}`);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const fetchAuditLogsByMemberId = async (
  memberId: number,
  limit: number = 50,
  offset: number = 0
) => {
  try {
    const response = await axios.get(`/audit-logs/member/${memberId}`, {
      params: { limit, offset },
    });
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const fetchAuditLogsByResourceType = async (
  resourceType: string,
  limit: number = 50,
  offset: number = 0
) => {
  try {
    const response = await axios.get(`/audit-logs/resource/${resourceType}`, {
      params: { limit, offset },
    });
    return response.data;
  } catch (error) {
    throw error;
  }
};

// Alias for consistency
export const fetchAuditLogsApi = fetchAuditLogs;
